util/rowcodec/rowcodec_test.go

```diff
@@ -30,16 +29,9 @@ import (
 	"github.com/pingcap/tidb/util/codec"
 	"github.com/pingcap/tidb/util/collate"
 	"github.com/pingcap/tidb/util/rowcodec"
+	"github.com/stretchr/testify/require"
 )
 
-func TestT(t *testing.T) {
-	TestingT(t)
-}
-
-var _ = Suite(&testSuite{})
-
-type testSuite struct{}
-
 type testData struct {
```

Comment:
1. what's diffierence between `dt` and `bt`
2. Does `testData.def` mean default value ?

Reply:
I'd like to refer to the original author @lysu . Any idea on this snippet of code?

Reply:
1. `dt` should be "input" and `bt` should be "output"..sorry for use such short name 😭 

for example, some type like duration input will decode as flatten type -- `int` in this rowcodec level(but will unflatten later in little upper level

https://github.com/pingcap/tidb/blob/da5574a0678e809db3aeea2c8a250ae3011676ea/util/rowcodec/rowcodec_test.go#L426-L427

2. yes `def` means default value and only this case use it
3. 
https://github.com/pingcap/tidb/blob/da5574a0678e809db3aeea2c8a250ae3011676ea/util/rowcodec/rowcodec_test.go#L656

Reply:
Got it; Thanks

Reply:
@UBarney we can rename the variable to fix the readability in this pass, though - by pushing a dedicated commit.

Reply:
Ok, I'll do it

Reply:
how about we change `testData` to 
```
type testData struct {
	id           int64
	ft           *types.FieldType
	inputDatum   types.Datum
	expected     types.Datum
	defaultDatum *types.Datum
	handle       bool
}

```

Reply:
I'd prefer

```go
type testData struct {
	id           int64
	ft           *types.FieldType
	input   types.Datum
	output     types.Datum
	default *types.Datum
	handle       bool
}
```

we don't have to repeat the field type and keep input and output symmetric. An output from testdata is expected data obviously IMO.